app/models/pieces/piece.rb

```diff
@@ -15,15 +15,28 @@ def obstructed?(x_destination, y_destination)
     pieces = []
     x = x_position
     y = y_position
-
     if moving_horizontally?(y_destination)
-      x = (x_position + 1..x_destination - 1)
+      return false unless diff(x_destination, x_position) > 1
+      direction_start_x = x_destination <=> x_position
+      x_start = x_position + direction_start_x
+      x_end = x_destination - (diff(x_destination, x_position) > 1 ? direction_start_x : 0)
+      x = x_start < x_end ? x_start..x_end : x_end..x_start
```

Comment:
Ahh, last line I saw in the diff was 42 & I mismatched it as the end to the method. 

The intermediary variables were the way to go, I agree. There are a few ways to refactor this method for readability, but not worth it at this point IMO - the quicker merge will be more beneficial to the team